### PR TITLE
Added tool break speeds & more

### PR DESCRIPTION
**Pull request by ImDaBigBoss**
- Added tool break speeds for custom items, so that we don't need to rely on tags that Mojang seems to have abandoned.
- Added `texture_type` to the armor component for custom items.
- Added an option to force display custom items as handheld. This means that items that bedrock can display an item like java would with `"parent": "item/handheld"` in the item model.

### CODE REVIEW

**Comment by ImDaBigBoss** [comment]
I think there is something wrong with the break speeds on slabs. I've tested cobblestone slabs and they don't seem to be breaking correctly, even though they are in the break speeds list. Am I missing something?

File: api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/NonVanillaCustomItemData.java [hunk 1]
```diff
@@ -129,13 +136,6 @@ public interface NonVanillaCustomItemData extends CustomItemData {
      */
     boolean isHat();
 
-    /**
-     * Gets if the item is a tool. This is used to set the render type of the item, if the item is handheld.
-     *
-     * @return if the item is a tool
-     */
-    boolean isTool();
```
Comment (Camotoy) [inline api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/NonVanillaCustomItemData.java:137]: While I'm not aware of anyone using this, we can't break API like this without proper deprecation.

File: api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/NonVanillaCustomItemData.java [hunk 2]
```diff
@@ -169,8 +171,6 @@ interface Builder extends CustomItemData.Builder {
 
         Builder hat(boolean isHat);
 
-        Builder tool(boolean isTool);
```
Comment (Camotoy) [inline api/geyser/src/main/java/org/geysermc/geyser/api/item/custom/NonVanillaCustomItemData.java:172]: Same.

**Review by Camotoy — CHANGES_REQUESTED** [review]

**Comment by Camotoy** [comment]
Can you print an example output of the ItemComponentPacket with these changes?

**Comment by ImDaBigBoss** [comment]
[ItemComponentPacket.zip](https://github.com/GeyserMC/Geyser/files/9994588/ItemComponentPacket.zip)
Sorry it's a zip, GitHub won't take json.

**Comment by Kas-tle** [comment]
Note: do not attempt to run the build generated by this action as it will (likely) not work due to outdated mappings^

**Comment by Kas-tle** [comment]
Also it seems that we unfortunately still cannot fix https://github.com/GeyserMC/Geyser/issues/3348 as it seems the `use_efficiency` sub-component of `digger` is non-functional

**Comment by Kas-tle** [comment]
Given the changes that custom blocks required to get break speeds to function, I have implemented a fix for this in my custom blocks PR that basically implements server authoritative block breaking when a custom tool as used. See https://github.com/GeyserMC/Geyser/pull/3505. As such this would no longer be needed if that is merged. It also fixes the problem in more scenarios, as we can properly account for high level efficiency enchantments.

**Pull request closed without merging**